### ISSUE TITLE
Longer prints always fail (Lost communication with MCU?)

### ISSUE THREAD

**Issue by VanKurt** [KevinOConnor/klipper#2550]
Whenever I try to make alonger print (> 2hours), the print fails. The printer just stops.
In the klippy.log I don't see anything suspicious. The octoprint log however seems to be full of 

> | Recv: // Lost communication with MCU 'mcu'

which can't be good, right?
Is there any more info on what exactly is going wrong to be gained from the logs? I really would like to fix this thing :-)

[octoprint.log](https://github.com/KevinOConnor/klipper/files/4238850/octoprint.log)
[klippy.log](https://github.com/KevinOConnor/klipper/files/4238849/klippy.log)

**Comment by vladbabii** [comment]
1. Check power issues on the pi
2. Check power on the board
3. Usb cable can become loose due to vibrations - make sure it's held in place somehow on both ends
4. If you have a relay between the pi usb and the board, try removing it
5. If the usb cable is not shielded properly and you have other wires (especially bed or hotend heater) running paralell to it it can also cause it
6. If nothing seems to work, try a new/different sd card in the pi - if it has issues it can cause freezes

What board are you using and how is it connected?

**Comment by vladbabii** [comment]
Also, once you get that error, open a shell to the pi and run this command

```
dmesg
```

The last couple of lines will give you a linux error message if something broke on the usb connection - if there are any issues there it's not a klipper issue.

**Comment by KevinOConnor** [comment]
Also, see the FAQ: https://www.klipper3d.org/FAQ.html#i-keep-getting-random-lost-communication-with-mcu-errors

-Kevin

**Comment by VanKurt** [comment]
I'm currently trying to replace all parts in question (power supplies, cables etc). Let's see if that helps :-)
Since this is clearly not a Klipper issue, I'm closing this.

**Comment by VanKurt** [comment]
In the mean time I replaced a lot of stuff:

- The old Raspi 3B is gone, replaced with a new Pi 4
- New, official Raspberry 4 power supply
- New SD card
- Using latest OctoPi image (0.17 instead of 0.16)
- New USB to printer cable (also removed VCC pin from cable to make sure printer is only powered by its own supply, not via USB)

What I did not replace so far (things might get expensive):

- New Creality Ender 3 power supply
- New Creality Ender 3 mainboard

Still the print failed after 8 hours :-(

Before I do something stupid (like thow the printer out of the window or going back to marlin/SD card printing without Klipper and Octoprint), is there any more information to be gained from the logs?

Here are some more details:

- the display on the printer started to show some weird stuff after a few minutes. It seems to have gotten worse over time. Attached is a photo of the "final stage" this morning, when the print had failed.
![WeirdDisplay](https://user-images.githubusercontent.com/5222025/75317065-4e7e5780-5867-11ea-8ed7-e9793d90e9f7.jpg)
 - As recommended above, I called the dmesg command (see attached file)
[dmesg.txt](https://github.com/KevinOConnor/klipper/files/4253685/dmesg.txt)
- Also the latest klippy.log and octoprint.log are attached
[klippy.log](https://github.com/KevinOConnor/klipper/files/4253688/klippy.log)
[octoprint.log](https://github.com/KevinOConnor/klipper/files/4253690/octoprint.log)
- I have a USB Webcam attached to my PI. May that somehow cause trouble?

**Comment by Arksine** [comment]
You seem to be having the same problem reported in #1919. A search for the error in dmesg.txt reveals that it could potentially be a kernel issue.  Specifically,  raspberrypi/linux#1187 and raspberrypi/linux#2406 appear to be related. 

You could try disconnecting the webcam, or any of the solutions outlined in the issued above. 
 Alternatively, if you have a spare x86 pc/laptop sitting around, you could try using it as a dedicated Klipper host.

**Comment by vladbabii** [comment]
For what is worth, i never had an usb connectivity error on my printers connected to x86/x64 laptops. Only to PIs

**Comment by KevinOConnor** [comment]
I'm going to close this ticket as it looks like the conversation has concluded.  Many people successfully run Klipper on USB connections that stay up for multiple months - as far as we know, the failing USB connections on some printers is due to a hardware issue of some kind.

-Kevin

**Comment by joekochevar** [comment]
I'm also having this issue, but I noticed a common thread among a few of these, which is the plugin "theSpaghettiDetective," paired with USB webcams. I'm not positive, but I have a feeling that this plugin might be interacting poorly with Klipper.

I deactivated that plugin and downgraded my USB bus to 1.1 speeds and haven't had the issue in 26 hours and counting (where it previously failed at the 13 hour mark without fail). Will try reverting to USB 2.0 while keeping TSD deactivated next.
